### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'name' in check_super_with_arguments

### ISSUE THREAD

**Issue by tardyp** [PyCQA/pylint#3852]
### Steps to reproduce

run pylint with following file:

```
def patched_init(self, *args):
    super(Warning, self).__init__(*args)
Warning.__init__ = patched_init
```

### Current behavior

pylint should correctly return error about super with arguments

### Expected behavior

pylint crashes with traceback:

```
Traceback (most recent call last):
  File ".venv/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
  File ".venv/lib/python3.8/site-packages/pylint/__init__.py", line 22, in run_pylint
    PylintRun(sys.argv[1:])
  File ".venv/lib/python3.8/site-packages/pylint/lint/run.py", line 349, in __init__
    linter.check(args)
  File ".venv/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 862, in check
    self._check_files(
  File ".venv/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 896, in _check_files
    self._check_file(get_ast, check_astroid_module, name, filepath, modname)
  File ".venv/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 923, in _check_file
    check_astroid_module(ast_node)
  File ".venv/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1055, in check_astroid_module
    retval = self._check_astroid_module(
  File ".venv/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1100, in _check_astroid_module
    walker.walk(ast_node)
  File ".venv/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  File ".venv/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  File ".venv/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  [Previous line repeated 2 more times]
  File ".venv/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 72, in walk
    callback(astroid)
  File ".venv/lib/python3.8/site-packages/pylint/checkers/refactoring.py", line 732, in visit_call
    self._check_super_with_arguments(node)
  File ".venv/lib/python3.8/site-packages/pylint/checkers/refactoring.py", line 763, in _check_super_with_arguments
    or node.args[0].name != node_frame_class(node).name
```

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 21 2020, 10:48:26)
[Clang 11.0.3 (clang-1103.0.32.62)]
```

### analysis

It looks like node_frame_class(node) returns None, as we are not in a class.

I am not sure exactly what is the logic in this line, so not sure what fix to propose.

**Comment by hippo91** [comment]
@tardyp thanks for the report.
I can reproduce the bug.

**Comment by karlmolina** [comment]
As a workaround, I put the function inside of a new class and then disabled the bad-super-call error.

**Comment by hippo91** [comment]
@tardyp this issue should have been fixed by #3810. I cannot reproduce it anymore with `pylint`'s master. Thus i close this issue.

**Comment by tardyp** [comment]
thank you very much. Appreciate your work on pylint!!
